src/controllers/auth.controller.js

```diff
@@ -200,11 +226,128 @@ function resetPassword(req, res, next) {
         .catch(error => next(error));
 }
 
+/**
+ * Sends back 200 OK if a contactMethodVerificationToken is created
+ * @param req
+ * @param res
+ * @param next
+ * @returns {*}
+ */
+function dispatchVerificaitonRequest(req, res, next) {
+    // This expects an email, and a page url to construct the verification link. It
+    // uses `verifyAccountToken` to populate `messagingProtocol` token,
+    // auth expiration, and provider for a user (unless an invalid email is
+    // provided). Finally it uses nodemailer to dispatch an email with a
+    // verification link to the user.
+
+    // TODO: Better handle email construction.
+    // TODO: Explore notification-microservice based dispatcing of messages.
+    //       * JRB: This probably helps us verify via more protocols, such as SMS or
+    //              push notification.
+    const email = _.get(req, 'body.email');
+    const messagingProtocolVerifyPageUrl = _.get(req, 'body.messagingProtocolVerifyPageUrl');
+
+    if (!email) {
+        const err = new APIError('Invalid email', 'INVALID_EMAIL', httpStatus.BAD_REQUEST, true);
+        return next(err);
+    }
+    return User.verifyAccountToken(email, 3600)
+        .then((token) => {
+            const verificationLink = generateLink(messagingProtocolVerifyPageUrl, token);
+            const verificationDomain = messagingProtocolVerifyPageUrl.replace(/(^\w+:|^)\/\//, '').split('/')[0];
```

Comment:
Same comment here as the type of comment I made above about how `resetDomain` is derived from a regex, but what's really going on is not clear.

Please name the regex something like (I don't know, just brainstorming) `const extractDomainFromUrl = /(^\w+:|^)\/\//`

Reply:
Did it myself. Resolved.